### PR TITLE
Update Language Feature Status

### PR DESCRIPTION
**Pull request by jaredpar**
Reflecting our current status

### CODE REVIEW

File: docs/Language Feature Status.md [hunk 1]
```diff
@@ -11,22 +11,23 @@ efforts behind them.
...
+| [Extension GetEnumerator](https://github.com/dotnet/csharplang/issues/3194)                        | [features/extension-foreach](https://github.com/dotnet/roslyn/tree/features/extension-foreach)                 | [In Progress](https://github.com/dotnet/roslyn/issues/43184) | [YairHalberstadt](https://github.com/YairHalberstadt)                        | [333fred](https://github.com/333fred)                                              | [333fred](https://github.com/333fred) |
+| [Module initializers](https://github.com/RikkiGibson/csharplang/blob/module-initializers/proposals/module-initializers.md) | TBD                                                                                    | In progress / design                                         | [RikkiGibson](https://github.com/RikkiGibson) [jnm2](https://github.com/jnm2)| [AlekseyTs](https://github.com/AlekseyTs)                                          | [gafetr](https://github.com/gafter) |
```
Comment (alrz) [inline docs/Language Feature Status.md:30]: > `[gafetr]`

typo

Comment (jaredpar) [reply docs/Language Feature Status.md:30]: hahah ... and he approved the PR :smile:

Comment (jaredpar) [inline docs/Language Feature Status.md:30]:
```suggestion
| [Module initializers](https://github.com/RikkiGibson/csharplang/blob/module-initializers/proposals/module-initializers.md) | TBD                                                                                    | In progress / design                                         | [RikkiGibson](https://github.com/RikkiGibson) [jnm2](https://github.com/jnm2)| [AlekseyTs](https://github.com/AlekseyTs)                                          | [gafter](https://github.com/gafter) |
```

**Comment by jcouv** [comment]
@jaredpar FYI, I made a small tweak to your PR, to add link to module initializers test plan (in the "status" column)

**Pull request merged**